src/libstd/sys/unix/os.rs

```diff
@@ -204,6 +215,21 @@ pub fn load_self() -> Option<Vec<u8>> {
     }
 }
 
+#[cfg(target_os = "openbsd")]
+pub fn load_self() -> Option<Vec<u8>> {
+    extern {
+        fn __load_self() -> *const c_char;
```

Comment:
Typically any rust-provided functions are also prefixed with `rust`, so `rust_load_self` is probably fine here.